### PR TITLE
[CFE-628]: Refactor(types) - custom minitia chain config

### PR DESCRIPTION
**Pull request by Poafs1**
## Summary by CodeRabbit

- **New Features**
	- Enhanced validation logic for network configuration and gas fee details.
	- Introduced new naming conventions for properties to improve clarity and consistency.
	- Added new schemas for input validation using the `zod` library.

- **Bug Fixes**
	- Resolved inconsistencies in naming conventions across various components.

- **Refactor**
	- Standardized naming conventions for gas pricing and network configuration properties.
	- Improved modularity and readability of validation schemas.

- **Documentation**
	- Updated documentation to reflect changes in naming conventions and validation logic.

### CODE REVIEW

File: src/lib/types/chainConfig.ts [hunk 1]
```diff
@@ -163,6 +176,11 @@ const zRegistry = z.object({
   assets: zAsset.array(),
 });
 
+export const zGas = z.object({
+  gasAdjustment: zNumberInput,
+  maxGasLimit: zNumberInput,
+});
+
 export const zApiChainConfig = z
```
Comment (songwongtp) [inline src/lib/types/chainConfig.ts:184]: let change to `zChainConfig`

File: src/lib/pages/custom-network/types.ts [hunk 2]
```diff
@@ -202,209 +301,131 @@ export const zAddNetworkManualChainConfigJson = ({
...
-      })),
-    },
-  }));
+      },
+    })
+  );
 
 export type AddNetworkManualChainConfigJson = z.infer<
   ReturnType<typeof zAddNetworkManualChainConfigJson>
 >;
```
Comment (songwongtp) [inline src/lib/pages/custom-network/types.ts:414]: // MARK: AddNetworkJsonChainConfigJson

File: src/lib/pages/custom-network/types.ts [hunk 3]
```diff
@@ -202,209 +301,131 @@ export const zAddNetworkManualChainConfigJson = ({
...
-      move:
-        val.vmType === VmType.MOVE
-          ? {
-              enabled: true,
-              moduleMaxFileSize: 1_048_576,
-            }
-          : { enabled: false },
-      pool: {
-        enabled: false,
+  }).transform<ChainConfig>(
```
Comment (songwongtp) [inline src/lib/pages/custom-network/types.ts:304]: // MARK: AddNetworkManualChainConfigJson

File: src/lib/pages/custom-network/types.ts [hunk 4]
```diff
@@ -2,409 +2,436 @@ import type { ChainConfig } from "@alleslabs/shared";
...
+    const { name, base, symbol } = val;
+
+    if (!/^[a-z-]+$/.test(name))
+      ctx.addIssue({
+        code: ZodIssueCode.custom,
+        message: "Alphabet (a-z) or hyphen (-) only",
+        path: ["name"],
+      });
+
+    if (!/^[a-z0-9/:._-]+$/.test(base))
```
Comment (evilpeach) [inline src/lib/pages/custom-network/types.ts:221]: Let's create internal function to handle this.

File: src/lib/pages/custom-network/types.ts [hunk 5]
```diff
@@ -2,409 +2,436 @@ import type { ChainConfig } from "@alleslabs/shared";
...
-      invalid_type_error: mustBeNumbersOnly,
-    })
-    .int({
+// MARK: WalletRegistryForm
+const zWalletRegistryAssetDenom = zDenomUnit
+  .pick({ denom: true, exponent: true })
+  .superRefine((val, ctx) => {
+    const { denom, exponent } = val;
+
+    if (!/^[a-z0-9/:._-]+$/.test(denom))
```
Comment (evilpeach) [inline src/lib/pages/custom-network/types.ts:185]: The `denom` is tested in the same way. Let's use utility function.

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**